packages/code-studio/src/dashboard/linker/Linker.tsx

```diff
@@ -248,21 +321,21 @@ class Linker extends Component {
         overrideIsolatedLinkerPanelId ?? isolatedLinkerPanelId
       );
 
-      if (type === LinkType.INVALID) {
+      if (type === 'invalid') {
         log.debug('Ignore invalid link connection', linkInProgress, end);
         return;
       }
 
-      // FILTER_SOURCE links have a limit of 1 link per target
+      // filterSource links have a limit of 1 link per target
       // New link validation passed, delete existing links before adding the new one
-      if (type === LinkType.FILTER_SOURCE) {
+      if (type === 'filterSource') {
         const { links } = this.props;
         const existingLinkPanelId = isReversed ? start.panelId : end.panelId;
```

Comment:
This might be safer in a switch/case statement since TS will check it exhaustively. If we were to add a new link state for any reason, in theory it should immediately catch that it's unhandled here (or caught by a default)